mhs/inbound/main.py

```diff
@@ -60,15 +60,17 @@ def load_party_key(data_dir: pathlib.Path) -> str:
     return party_key
 
 
-def start_inbound_server(certs_file: str, key_file: str, party_key: str,
+def start_inbound_server(local_certs_file: str, ca_certs_file: str, key_file: str, party_key: str,
                          workflows: Dict[str, workflow.CommonWorkflow],
                          persistence_store: persistence_adaptor.PersistenceAdaptor
                          ) -> None:
     """
 
     :param persistence_store: persistence store adaptor for message information
-    :param certs_file: The filename of the certificate to be used to identify this MHS to a remote MHS.
-    :param key_file: The filename of the private key for the certificate identified by certs_file.
+    :param local_certs_file: The filename of the certificate to present for authentication.
+    :param ca_certs_file: The filename of the CA certificates that form the certificate chain for the certificate
```

Comment:
A minor point, but I don't think these certificates form the cert chain for the local certificate. Rather, I think they're used as the CA certificates store to verify the certificate presented by Spine when a connection is made to the inbound service.

This would seem to be backed up by the [documentation for `load_verify_locations`](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_verify_locations), which says:
>Load a set of “certification authority” (CA) certificates used to validate other peers’ certificates [...]

Reply:
I've taken a look on the command line at one of our Opentest certs. They do form a cert chain in terms of being signed by each other, with the root cert being self-signed.

I see what the Python docs say, that said the docs for [`load_cert_chain`](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_cert_chain) say:
>  The certfile string must be the path to a single file in PEM format containing the certificate as well as any number of CA certificates needed to establish the certificate’s authenticity.

So you could argue that we shouldn't just be passing the endpoint certificate there. I think the CA certs we're passing to `load_verify_locations` are being used to verify certs on both sides of the connection.

To try and understand this better, I've taken a quick dive through the implementation of `load_verify_locations`. This function essentially ends up calling [`SSL_CTX_load_verify_locations`](https://github.com/python/cpython/blob/69b3718b183a698202bd67488639bffd64a488bc/Modules/_ssl.c#L4053)  which is documented [here](https://www.openssl.org/docs/man1.0.2/man3/SSL_CTX_load_verify_locations.html). I'm more confused by reading these docs, they kind of sound like they back up my hypothesis, but they also sound like they don't .

@gareth-allan what do you think?

Reply:
I think it's a difference in what these certificates are used for. The cert file passed to `load_cert_chain` is what's presented when a connection is made to this application, so it _could_ be a chain of certificates, if that's what's needed to allow the remote end of the connection to validate our certificate, but that's not necessary with the infrastructure we're using.

The CA certs passed to `load_verify_locations` are used to validate the certificate presented by the remote end of the connection. In our case, we use the sub CA and the root CA certs, so we can establish the certificate chain if a lone certificate is presented.

All that said, I don't think we can hope to explain the whole situation in a docstring, and I'm not keen to hold up this review over it. Could we just have our docstrings say something like "as passed to SSLContext.load_verify_locations()"?